vloebel/back2work
# Routes/vicky2

**THIS UPDATE CONTAINS THE COMPLETED USER AND MEETINGS ROUTES** – it does not include the routes that will let us assign people as participants in meetings, but hopefully that will be ready by tomorrow morning.
**THE FILE Resources/backendSpec.html  has a graphic list of each route that has been implemented** (and the ones that haven’t) together with information on how to access the routes in insomnia.   
  For example, to add a user you use POST localhost:3002/api/users
**It would be good for someone else to test all this asap (doesn’t take long)  so we can pull to develop and not pile up too many changes waiting approval.**
TO GET THE FILES TO TEST:
•	on your local repo, type git fetch  (or pull…? I think either works)
•	change to this test branch with “git checkout routes/vicky2” – it should load that directory for you even if you don’t see it before typing the checkout command
THIS UPDATE CONTAINS CHANGES TO THE SCHEMA AND RELATONSHIPS - In order to run, you must:
1.	Open our server file program: server.js and change force to TRUE:
 sequelize.sync({ force: true})
2.	Start mySql on the command line and rebuild the schema: 
source: ./db/schema.sql
3.	Quit mySQL and run our server “npm start” – this will REBUILD the database and tables, but they will be empty
4.	Make sure you restore force:false in server.js because otherwise it will wipe the database every time you restart. (been there, done that) 
THERE IS SOME TEST DATA IN THE FILE db/insomniaTestData.txt
5.	Test the back ind in insomnia by picking the route you want and copying one object (one meeting or one user) at a time and pasting them into insomnia
THE FILE Resources/backendSpec.html has a list of all the routes that are currently implemented and/or planned, along with the routes to get to them, so you can see what to do. (Vicky will also talk you through it!)
IF YOU WANT TO USE THE SEEDS FILE JUST TO HAVE A BIT FULLER DATABASE there are some more steps. 
6.	open mySql again and seed the database with “source ./db/seeds.sql “  YOU HAVE to rebuild and sync the database by running the server after step 2 and before here, because otherwise there won’t be any tables to seed.
7.	quit mySQL.  
8.	Make sure you followed step 4 above and that force:false is set in server.js (and that it’s saved) because otherwise starting the server will wipe the database.
9.	start the server. You still need to copy;paste into inquirer to test the routes with new data, but at least they won’t be all alone in the database.
